src/main/java/pl/byczazagroda/trackexpensesappbackend/dto/RscpDTO.java

```diff
@@ -0,0 +1,18 @@
+package pl.byczazagroda.trackexpensesappbackend.dto;
+
+import org.hibernate.annotations.Immutable;
+import org.springframework.lang.NonNull;
+import org.springframework.lang.Nullable;
+import pl.byczazagroda.trackexpensesappbackend.rscp.RscpStatus;
+
+/**
+ * {@code RscpDTO} record used to transfer date between service and controller. Contain a status, message and
+ * body.
+ *
+ * @param status  the status code
+ * @param message the status message
+ * @param body    the response body
+ */
+@Immutable
+public record RscpDTO<T>(@NonNull RscpStatus status, @Nullable String message, @Nullable T body) {
+}
```

Comment:
Co znaczy skrót Rscp?

Reply:
**R**esponse **S**ervice to **C**ontroller **P**rotocol.